webpack/assets/javascripts/react_app/components/common/forms/ForemanForm/ForemanForm.js

```diff
@@ -18,42 +25,71 @@ export const prepareErrors = errors =>
 export const isInitialValid = ({ validationSchema, initialValues }) =>
   !validationSchema ? true : validationSchema.isValidSync(initialValues);
 
-const ForemanForm = props => (
-  <Formik
-    onSubmit={(values, actions) =>
-      props.onSubmit(values, actions).catch(exception => {
-        actions.setSubmitting(false);
-        actions.setErrors(prepareErrors(exception.errors));
-      })
-    }
-    initialValues={props.initialValues}
-    validationSchema={props.validationSchema}
-    isInitialValid={isInitialValid}
-    enableReinitialize={props.enableReinitialize}
-  >
-    {formProps => {
-      const disabled = formProps.isSubmitting || !formProps.isValid;
+const ForemanForm = ({
+  onSubmit,
+  onSuccess,
+  children,
+  initialValues,
+  validationSchema,
+  enableReinitialize,
+  onCancel,
+}) => {
+  const [key, setKey] = useState(undefined);
+  const error = useSelector(state => selectAPIError(state, key));
+  const status = useSelector(state => selectAPIStatus(state, key));
+  const [formActions, setFormActions] = useState();
 
-      const submissionError = formProps.errors._error;
-      return (
-        <Form
-          onSubmit={formProps.handleSubmit}
-          onCancel={props.onCancel}
-          disabled={disabled}
-          error={submissionError}
-          errorTitle={
-            submissionError && submissionError.severity === 'danger'
-              ? __('Error! ')
-              : __('Warning! ')
-          }
-          submitting={formProps.isSubmitting}
-        >
-          {cloneChildren(props.children, { formProps, disabled })}
-        </Form>
+  useEffect(() => {
+    debugger;
+    if (key && formActions && error) {
+      formActions.setSubmitting(false);
+      formActions.setErrors(
+        prepareErrors(error?.response?.data?.error?.errors)
       );
-    }}
-  </Formik>
-);
+      setKey(undefined);
```

Comment:
do we really need to keep track of the key here? IMO after the modal disappear, this component is gone
so no need to track the key I think

Reply:
not anymore :)
the key is passing directly to `ForemanForm`